### PR TITLE
Motech-2423 Added IT to check cascade delete

### PR DESCRIPTION
**Pull request by shanmukhm**
To share with Sebastian

### CODE REVIEW

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

File: platform/mds/mds-test-bundle/src/main/java/org/motechproject/mds/test/domain/cascadedelete/EntityY.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import org.motechproject.mds.annotations.Cascade;
+import org.motechproject.mds.annotations.Entity;
+import org.motechproject.mds.annotations.Field;
+import org.motechproject.mds.domain.MdsEntity;
+
+@Entity
+public class EntityY extends MdsEntity{
+    @Field
+    private Set<EntityX> x;
```
Comment (sebbrudzinski) [inline platform/mds/mds-test-bundle/src/main/java/org/motechproject/mds/test/domain/cascadedelete/EntityY.java]: You may try changing this to just EntityX (one to many relationship)

File: platform/mds/mds-test-bundle/src/test/java/org/motechproject/mds/test/osgi/MdsDdeBundleIT.java [hunk 2]
```diff
@@ -470,6 +488,51 @@ public void testSetOfEnumsAndSetOfStrings() {
...
+        EntityY y2 = new EntityY();
+        
+        x1.getY().add(y1);
+        x1.getY().add(y2);
+        y2.getX().add(x1);
+        y2.getZ().add(z2);
+        y1.getX().add(x1);
+        y1.getZ().add(z1);
+        
+        entityXDataService.create(x1);
```
Comment (sebbrudzinski) [inline platform/mds/mds-test-bundle/src/test/java/org/motechproject/mds/test/osgi/MdsDdeBundleIT.java]: Creating x1 automatically creates all its relationships so further creates should be removed.

File: platform/mds/mds-test-bundle/src/test/java/org/motechproject/mds/test/osgi/MdsDdeBundleIT.java [hunk 3]
```diff
@@ -470,6 +488,51 @@ public void testSetOfEnumsAndSetOfStrings() {
...
+        entityYDataService.create(y1);
+        entityYDataService.create(y2);
+        entityZDataService.create(z1);
+        entityZDataService.create(z2);
+        
+        EntityX testX = entityXDataService.findById(x1.getId());
+        assertNotNull(testX);
+        
+        EntityY testY = entityYDataService.findById(y1.getId());
```
Comment (sebbrudzinski) [inline platform/mds/mds-test-bundle/src/test/java/org/motechproject/mds/test/osgi/MdsDdeBundleIT.java]: Unnecessary. We should rather assert that x instance has got its y relationships set and that the y instances from x have got their z relationships set.

**Pull request closed without merging**